src/alchemlyb/parsing/amber.py

```diff
@@ -65,8 +65,10 @@ def __init__(self, filename):
         self.filename = filename
         try:
             self.fileh = anyopen(self.filename, 'r')
-        except Exception as ex:  # pragma: no cover
-            logger.exception("Cannot open file %s", filename)
+        except Exception as ex:
+            msg = f"Cannot open file {filename}"
+            logger.exception(msg)
+            raise FileNotFoundError(msg) from ex
```

Comment:
Why on earth did we not even raise the exception there... ? Thanks for raising.

However, it's not clear that this is a FileNotFoundError. Let's not pretend that we know what went wrong. Just reraise the original exception
```suggestion
            raise
```

Reply:
Done!